pkg/beacon/beacon.go

```diff
@@ -24,35 +24,36 @@ var logger = log.Logger("keep-beacon")
 // otherwise enters a blocked loop.
 func Initialize(
 	operatorPublicKey *operator.PublicKey,
+	beaconChainV1 beaconchain.Interface, // TODO: Remove once transition from v1 to v2 is done.
 	beaconChain beaconchain.Interface,
```

Comment:
This is an identical solution as in #3077 but we'll have conflicts. I suggest we merge #3077 first to have the client register in the pool. Then merge `main` to this PR.